src/main/java/org/jetbrains/bsp/bazel/server/bsp/resolvers/actiongraph/ActionGraphV2Parser.java

```diff
@@ -0,0 +1,109 @@
+package org.jetbrains.bsp.bazel.server.bsp.resolvers.actiongraph;
+
+import com.google.common.collect.Lists;
+import com.google.devtools.build.lib.analysis.AnalysisProtosV2;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.jetbrains.bsp.bazel.commons.Uri;
+
+public class ActionGraphV2Parser implements ActionGraphParser {
+  private final AnalysisProtosV2.ActionGraphContainer actionGraph;
+  private final Map<Integer, AnalysisProtosV2.PathFragment> pathFragmentMap;
+
+  public ActionGraphV2Parser(AnalysisProtosV2.ActionGraphContainer actionGraph) {
+    this.actionGraph = actionGraph;
+    pathFragmentMap =
+        actionGraph.getPathFragmentsList().stream()
+            .collect(Collectors.toMap(AnalysisProtosV2.PathFragment::getId, fragment -> fragment));
+  }
+
+  @Override
+  public List<String> getInputsAsUri(String target, String execRoot) {
```

Comment:
this method looks the same for both versions, am I right?

Reply:
Extracted to interface